### PR TITLE
CI: Speed up OpenTitan parsing jobs

### PR DESCRIPTION
**Pull request by mglb**
Parsing test has been parallelized and is executed in 2 similar jobs.
All test cases are grouped in following groups:

| Group          | Description                                                                                               |
|----------------|-----------------------------------------------------------------------------------------------------------|
| SKIPPED_CORES  | Never tested cores. Always assumed to pass.                                                               |
| PASSING_CORES  | Cores initially assumed to pass testing. Tested only when any of its users failed a test.                 |
| TESTED_CORES   | Cores tested normally. Marked without testing as `dependency-failed` when any of its dependencies failed. |
| TOP_DOWN_CORES | Similar to PASSING_CORES, but uses `test-pass` and `test-fail` as its statuses.                           |

## Quick test

Runtime: few minutes

Markdown summary: https://github.com/antmicro/yosys-systemverilog/actions/runs/4542259935/attempts/1#summary-12330837198
Terminal summary (logs and stuff): https://github.com/antmicro/yosys-systemverilog/actions/runs/4542259935/jobs/8005547681?pr=1607#step:11:361

## Full test

Runtime: ~2h

This is mostly a backup. Quick test should catch all meaningful status changes, but bugs happen.
This test starts from the top core and goes down.

Markdown summary: https://github.com/antmicro/yosys-systemverilog/actions/runs/4542259935/attempts/1#summary-12330836828
Terminal summary: https://github.com/antmicro/yosys-systemverilog/actions/runs/4542259935/jobs/8005547395?pr=1607#step:11:353

### CODE REVIEW

File: uhdm-tests/opentitan/opentitan-parsing-test/ot-cores-passlist.txt [hunk 1]
```diff
@@ -1,27 +1,22 @@
...
-lowrisc:dv:sw_test_status:0
 lowrisc:dv_verilator:memutil_dpi:0
 lowrisc:dv_verilator:memutil_dpi_scrambled:0
-lowrisc:dv_verilator:memutil_verilator:0
-lowrisc:dv_verilator:simutil_verilator:0
 lowrisc:ibex:ibex_core:0.1
 lowrisc:ibex:ibex_icache:0.1
 lowrisc:ibex:ibex_pkg:0.1
 lowrisc:ibex:ibex_top:0.1
+lowrisc:ip:alert_handler_component:0.1
```
Comment (kamilrakoczy) [inline uhdm-tests/opentitan/opentitan-parsing-test/ot-cores-passlist.txt:15]: Why this file changes?

**Review by kamilrakoczy — APPROVED** [review]
Could you please explain why passlist changes? Other than that, LGTM.

File: uhdm-tests/opentitan/opentitan-parsing-test/ot-cores-passlist.txt (hunk 1 above)
Comment (kamilrakoczy) [reply uhdm-tests/opentitan/opentitan-parsing-test/ot-cores-passlist.txt:15]: Makes sense. Thanks for the explanation.

**Review by wsipak — COMMENTED** [review]
Does this mean we no longer need the OT targets in `uhdm-tests/opentitan/Makefile` and the `uhdm-tests/iterate_cores.py` file?
We could get rid of these if they're no longer used.
Apart from this, LGTM.

**Comment by mglb** [comment]
@wsipak done.
I've also rebased the branch.

**Review by wsipak — APPROVED** [review]

**Pull request merged**